packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -485,14 +485,50 @@ public void uploadAsync(final String fileUriString, final String url, final Map<
         }
       }
 
-      RequestBody body = RequestBody.create(null, uriToFile(fileUri));
-
+      File file = uriToFile(fileUri);
       String method = "POST";
       if (options.containsKey("httpMethod")) {
         method = importHttpMethod((int) (double) options.get("httpMethod"));
       }
 
-      requestBuilder.method(method, body);
+      int type = 0;
+      if (options.containsKey("uploadType")) {
+        type = (int) (double) options.get("uploadType");
```

Comment:
What's this 😄?
Cannot we get is somehow differently?

Reply:
I don't like it too. However, when we pass `number` from js, it's stored as a `Double`.  Jave doesn't allow conversion like this:
```
Double -> int (where Double is a class). 
```
So, we need to unbox it and then we can safely convert to int.